### PR TITLE
feat: allow jsx element as error or hint prop

### PR DESCRIPTION
**Pull request by sampolahtinen**
This PR introduces a change where TextField and Hint component expands the `error` and `hint` prop  type for allowing a jsx element to be passed as a value, allowing for more customized error and hint messages.
 
Closes CP-2595

### CODE REVIEW

File: src/components/text-field/TextField.tsx [hunk 1]
```diff
@@ -16,10 +16,17 @@ export type TextFieldProps = {
...
+  hint?: string | JSX.Element
+  /**
+   * An optional error to show next to the TextField. If a `validator` is also supplied, the `validator` takes precendence
+   *
+   * Allowing string or react element for more customized use cases.
+   * Such as providing a guiding link to support the error.
+   * Furthermore, with custom components, more complex
+   * accessibility requirements can be implemented.
+   **/
+  error?: string | JSX.Element
```
Comment (sampolahtinen) [inline src/components/text-field/TextField.tsx:29]: I did not want to add any TODO comments here as this is a public facing library.

Comment (tatianainama) [reply src/components/text-field/TextField.tsx:29]: Not sure about the extra explanation here, I would omit it: I think the types are speaking for themselves 😄 
@luanvdw wdyt?

Comment (sampolahtinen) [reply src/components/text-field/TextField.tsx:29]: I will remove this. 

I recalled us talking about leaving a comment to either lens or pdp, but I feel that lens is not the place for it.

Ill add a comment in PDP describing that this behavior might chance as a result of upcoming Design System proposal

Comment (luanvdw) [reply src/components/text-field/TextField.tsx:22]: @tatianainama @sampolahtinen I agree (with the already removed comments) I feel that neither Lens nor PDP is the right place to _really_ document component lib shortfalls. Lens should advocate for its current state (even if not ideal) and PDP should use what is available from Lens and at the same time we know that an implementation can be improved upon/refined. I have created a page for us to better document scenarios like this, that I have shared with you.

**Pull request merged**